db/migrate/20221026070600_add_user_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, foreign_key: true
```

Comment:
- [ ] Posts should have an `author_id`, not `user_id`. In that case, this **references** definition should point to the `author`, not the `user`. You can point Rails to the user using the `to_table` attribute of `foreign_key`. Kindly look at this [Stackoverflow](https://stackoverflow.com/a/42056089) for hint.